lib/src/solver/report.dart

```diff
@@ -375,32 +412,57 @@ $contentHashesDocumentationUrl
         maxAge: Duration(days: 3),
       );
 
+      var messagePrefix = '(';
+      var hasAdvisory = false;
+      if (status.advisoriesUpdated != null) {
+        final advisories = await id.source.getAdvisoriesForPackageVersion(
+          id,
+          _cache,
+          Duration(days: 3),
+        );
+        hasAdvisory = advisories != null && advisories.isNotEmpty;
+        if (hasAdvisory) {
+          final advisoryFootnotes = <int>[];
+          for (final adv in advisories) {
+            advisoryFootnotes.add(advisoriesMap.length);
+            advisoriesMap[advisoriesMap.length] = adv.id;
+          }
+
+          messagePrefix =
+              '(${_constructAdvisoriesMessage(advisoryFootnotes)}, ';
+        }
+      }
+
       if (status.isRetracted) {
         if (newerStable) {
           message =
-              '(retracted, ${maxAll(versions, Version.prioritize)} available)';
+              '${messagePrefix}retracted, ${maxAll(versions, Version.prioritize)} available)';
         } else if (newId.version.isPreRelease && newerUnstable) {
-          message = '(retracted, ${maxAll(versions)} available)';
+          message = '${messagePrefix}retracted, ${maxAll(versions)} available)';
         } else {
-          message = '(retracted)';
+          message = '${messagePrefix}retracted)';
         }
       } else if (status.isDiscontinued &&
           [DependencyType.direct, DependencyType.dev]
               .contains(_rootPubspec.dependencyType(name))) {
         if (status.discontinuedReplacedBy == null) {
-          message = '(discontinued)';
+          message = '${messagePrefix}discontinued)';
         } else {
           message =
-              '(discontinued replaced by ${status.discontinuedReplacedBy})';
+              '${messagePrefix}discontinued replaced by ${status.discontinuedReplacedBy})';
         }
       } else if (newerStable) {
         // If there are newer stable versions, only show those.
-        message = '(${maxAll(versions, Version.prioritize)} available)';
+        message =
+            '$messagePrefix${maxAll(versions, Version.prioritize)} available)';
       } else if (
           // Only show newer prereleases for versions where a prerelease is
           // already chosen.
           newId.version.isPreRelease && newerUnstable) {
-        message = '(${maxAll(versions)} available)';
+        message = '$messagePrefix${maxAll(versions)} available)';
+      } else if (hasAdvisory) {
+        message =
```

Comment:
This is a bit funky.
How about having a list of notes that you join with a ','.

```
final notes = <String>[];
final advisories = await await id.source.getAdvisoriesForPackageVersion(id, _cache, Duration(days: 3));
if (advisories.isNotEmpty) {
  int start = advisoryFootnotes.length;
  advisoryFootnotes.addAll(advisories);
  int end = advisoryFootnotes.length;
  notes.add(_constructAdvisoryMessage(start, end));
}
if (status.isRetracted) {
  notes.add('retracted');
  if (newerStable) {
    notes.add('${maxAll(versions, Version.prioritize)} available';
  } else if
  ...
} else if (status.isDiscontinued) {
  if (status.discontinuedReplacedBy == null) {
     notes.add('discontinued');
  } else {
     notes.add('discontinued replaced by ${status.discontinuedReplacedBy}');
  }
}
message = notes.isEmpty ? '' : '(${notes.join(', '})';
```

Reply:
Much better. Done, except that the message is null when notes is empty.